### ISSUE TITLE
Make sure that Mint tests can be executed even with self-signed certs

### ISSUE THREAD

**Issue by aweisser** [iternity-dotcom/minio#1]
**Is your feature request related to a problem? Please describe.**
We have problems running the Mint test suite against self signed SSL certs.

**Describe the solution you'd like**
An option for the Mint Test-Suite to accept self-signed certs.

**Describe alternatives you've considered**
Working around it maybe possible by editing the /etc/hosts?

**Additional context**
There is a closed (unresolved) issue in the old minio/mint repo.
Not sure why the contributers marked this one as "not needed".
https://github.com/minio/mint/issues/221

**Comment by aweisser** [comment]
@rluetzner Can you verify the workaround with /etc/hosts first? If this doesn't work than we can maybe assign the task to @andrebruch

**Comment by rluetzner** [comment]
@aweisser , I was unable to get this to work by modifying /etc/hosts. Generating a certificate with a valid hostname made a lot more tests possible, but not all test tools and SDKs honor the certificate trust established in the mint.sh script.

**Comment by iternity-dotcom** [comment]
@rluetzner So, should we change something here?

**Comment by rluetzner** [comment]
No, sorry. I finally got it running. The actual problem was that I was using a self-signed certificate for a different hostname ('localhost') than I used to access the server. After generating a fitting self-signed certificate most of the tests run.

There are additional issues, e.g.
1. Mint executes 'server side encryption' tests that fail, because the connection between a reverse proxy and the MinIO instance is no longer encrypted and
2. a single failing test causes all following tests to be skipped (what I call 'short-circuit behavior').

We should however track these in additional issues and close this one.
